IGNITE-5452

BiryukovVA opened the pull request.

sboikov committed:
Fixed exception conversion in DataStreamerImpl.

akuznetsov-os committed:
GG-12031 Reworked status command.

gvvinblade committed:
IGNITE-5516 enable tests

shroman committed:
IGNITE-5492: Fixed LOCAL cache metrics. This closes #2133.

sergey-chugunov-1985 committed:
IGNITE-5519 test restored

Commit:
Added missing header.

Commit:
Merge problem fix: unused field removal

ilantukh committed:
Fixed cache count checks in IgniteStandByClusterSuite tests.

ilantukh committed:
Merge branch 'ignite-2.1.1' of https://github.com/gridgain/apache-ignite into ignite-2.1.1

nva committed:
IGNITE-5366 Fixed tables position in modals.

Klaster1 committed:
IGNITE-5271 Use same blank cluster for basic and advanced configuration.

dgovorukhin committed:
ignite-gg-12301 fix Activate/Deactivate test fail

akuznetsov-os committed:
IGNITE-5461 Added collecting of memory metrics.

dgovorukhin committed:
ignite-2.1.1 add test dependency

dgovorukhin committed:
Merge remote-tracking branch 'professional/ignite-2.1.1' into ignite-2.1.1

Commit:
WAL history size: code commenting

dgovorukhin committed:
ignite-gg-12301 fix assert message

dgovorukhin committed:
Merge remote-tracking branch 'professional/ignite-2.1.1' into ignite-2.1.1

Commit:
IGNITE-5524: Fixed IgniteComputeImpl deserialization.

isapego committed:
IGNITE-5525: Fix for ODBC example

(cherry picked from commit 7cfad49)

AMashenkov committed:
IGNITE-5527: Prevent starvation in stripe pool on unstable topology.

(cherry picked from commit 97ea507)

Commit:
GG-12317 Restore snapshot functionality after split GridCacheDatabaseSharedManager and GridCacheSnapshotManager
GG-12318 Remove possibility to edit snapshot by deletion certain cache from snasphot

dgovorukhin committed:
Merge remote-tracking branch 'professional/ignite-gg-12289' into ignite-2.1.1

dgovorukhin committed:
ignite-gg-12288 fix SnapshotUtilityTestWithDaemonInTopologyTest, send compute activate from daemon node

dgovorukhin committed:
ignite-gg-12301 fix assert message

kdudkov committed:
Fixed serialization of conflict version.

devozerov committed:
Trying to fix compilation error.

devozerov committed:
JavaDoc fix.

ilantukh committed:
Fixed incorrect handling of DS in cache not present on local node.

glukos committed:
ignite-5226 Cleanup client topology on cache group stop

Jokser committed:
ignite-5503 New compare topology versions for dhtPartMap comparison

sboikov committed:
ignite-5512 enabled test

dgovorukhin committed:
cleanup memPlcMap on stop + add restart test

Commit:
IGNITE-5301: Fixed invalid memcpy parameter ordering leading to JVM crash in H2TreeIndex. This closes #2029. This closes #2151.

ilantukh committed:
ignite-5519 : Removed failing assertion. Added assertions for safety of concurrent modifications.

ilantukh committed:
Merge branch 'ignite-5519' of https://github.com/gridgain/apache-ignite into ignite-5519

ilantukh committed:
Merge branches 'ignite-2.1.1' and 'ignite-gg-12325' of https://github.com/gridgain/apache-ignite into ignite-2.1.1

akuznetsov-gridgain committed:
ignite-2.1 Fixed UnsupportedOperationException in case of single file.

Commit:
Fixed NPE inf GridCacheInternalKeyImpl.

Commit:
Merge remote-tracking branch 'upstream/ignite-2.1.1' into ignite-2.1.1

ilantukh committed:
ignite-5519 : Muted test with link to another ticket - ignite-5529.

ilantukh committed:
Merge branch 'ignite-2.1.1' of https://github.com/gridgain/apache-ignite into ignite-5519

ilantukh committed:
GridCacheInternalKeyImpl - made groupName not-nullable.

dgovorukhin committed:
ignite-gg-12335 fix fail activation after fail get file lock

ilantukh committed:
Merge branches 'ignite-2.1.1' and 'ignite-5519' of https://github.com/gridgain/apache-ignite into ignite-2.1.1

dgovorukhin committed:
Merge branch 'ignite-gg-12335' into ignite-2.1.1

akuznetsov-gridgain committed:
IGNITE-5530 Fixed import indexes from RDBMS. Added systemSchemas to MySQLMetadataDialect. Fixed POJO hashCode generation for double.

devozerov committed:
IGNITE-5531: Fixed hang in Hibernate example (caused by not closed SessionFactory).

ilantukh committed:
Fixed initialization of PartitionCountersPage.

nva committed:
IGNITE-5415 Fixed memory policy configuration.

vsisko committed:
WEB-CONSOLE-2.1.1 Fixed variable name.

akuznetsov-os committed:
Added package-lock.json to .gitignore.

sboikov committed:
Do not need process missed handler error on stopping node.

sboikov committed:
Call GridDhtPartitionTopology.onExchange done for each group, not cache.

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

Commit:
IGNITE-5545: fix for Ignite PDS1: ignitePdsCheckpointSimulationTest 2 tests failed

sboikov committed:
Call GridDhtPartitionTopology.onExchange done for each group, not cache.

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

agoncharuk committed:
IGNITE-5528 - IS_EVICT_DISABLED flag is not cleared on cache store exception. This closes #2183.

dgovorukhin committed:
ignite-2.1.2 remove IgniteWalDirectoriesConfigurationTest as outdated

nva committed:
Web Console: Minor fix.

ptupitsyn committed:
IGNITE-5522 .NET: IgniteConfiguration.ActiveOnStart

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

Commit:
IGNITE-5545: fix for Ignite PDS1: ignitePdsCheckpointSimulationTest 2 tests failed, code commenting

Commit:
IGNITE-5545: fix for Ignite PDS1: ignitePdsCheckpointSimulationTest 2 tests failed, code commenting

Jokser committed:
ignite-gg-12322 Fix for cache restart

dgovorukhin committed:
ignite-gg-12329 do not update partition map, in case cluster transitioning to inactive state.

gvvinblade committed:
IGNITE-5548 Deadlock Detection uses IgniteCheckedException instead of TransactionTimeoutException

dgovorukhin committed:
ignite-5514 test is slow, break by timeout, unmute test

AMashenkov committed:
IGNITE-5552: ServiceProcessor recalculates all service assignments even if there is a pending topology change. This closes #2182.

dgovorukhin committed:
ignite-12336 fix npe in cluster state processor on stop daemon client node

dgovorukhin committed:
Merge branch 'ingite-gg-12329' into ignite-2.1.2

dgovorukhin committed:
Merge remote-tracking branch 'professional/ignite-gg-12336' into ignite-2.1.2

sergey-chugunov-1985 committed:
Storing binary metadata in FS

nva committed:
Merge branch 'web-console-2.1.1' into 'ignite-2.1.2'

Commit:
IGNITE-4467 Fixed ui for modal, select, grid. Added components: switcher, timepicker.

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

# Conflicts:
#	modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite2.java

sergey-chugunov-1985 committed:
New method to register marshaller mappings locally without triggering discovery messages exchange

glukos committed:
added javadoc to class header of GridNearGetRequest

glukos committed:
added javadoc to class header of GridNearGetRequest

Commit:
Code commenting after dev list questions

Klaster1 committed:
IGNITE-4467 Fix webconsole frontend build.

dgovorukhin committed:
ignite-gg-12324 fix caches start after restore hangs if concurrent transactions run

sk0x50 committed:
IGNITE-5076: improved multi-threaded start of nodes

Commit:
GG-12168 Do not store partitions state for local caches

dmitriyff committed:
IGNITE-4467 fixed table header scrolling with content

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheOffheapManagerImpl.java
#	modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite2.java

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheOffheapManagerImpl.java
#	modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite2.java

sboikov committed:
Quick fix for NPE in PageMemoryImpl.loadedPages

sboikov committed:
Added test for concurrent activation and node join.

dgovorukhin committed:
ignite-gg-12052 code comments

sergey-chugunov-1985 committed:
GG-12347 restoring binary metadata at correct moment of ObjectBinaryProcessor startup process

vsisko committed:
IGNITE-5577 Added missed fields to DTO.

nva committed:
IGNITE-5461 Support for Memory metrics and fixed cache metrics.

vsisko committed:
IGNITE-5599 Added check that node URL is valid. Added warning about ignite-rest-http.

sboikov committed:
Merge remote-tracking branch 'remotes/origin/master' into ignite-2.1.2

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtPartitionTopologyImpl.java

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

vsisko committed:
IGNITE-5536 Fixed Docker file generation.

sboikov committed:
ignite-2.1.2 Limit amount of debug logging

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

akuznetsov-os committed:
IGNITE-5599 Added log throttling.

akuznetsov-os committed:
IGNITE-4467 Added missing style.

sergey-chugunov-1985 committed:
fail added to IgnitePersistentStoreDataStructuresTest::testSet, link to JIRA is provided

mcherkasov committed:
Fixed "IGNITE-5521 Large near caches lead to cluster instability with metrics enabled". This closes #2201.

Signed-off-by: nikolay_tikhonov <ntikhonov@gridgain.com>

Klaster1 committed:
IGNITE-5611 Update some selects to new design.

Commit:
GG-12411 Backport [GG-12339] to 8.1.2
GG-12339 Improve logging of snasphot operation

sboikov committed:
ignite-2.1.2 detectLostPartitions minor optimization

Klaster1 committed:
IGNITE-5611 Fix text color cascading issue.

Klaster1 committed:
IGNITE-5627 Refactored grid columns menu.

akuznetsov-os committed:
Fixed CSS for IE11.

niktikhonov committed:
IGNITE-5424: GridServiceProxy does not unwraps exception message from InvocationTargetException. This closes #2168.

glukos committed:
GG-12407: Backport [GG-12364] to 8.1.2

mcherkasov committed:
IGNITE-5554 ServiceProcessor may process failed reassignments in timeout thread

mcherkasov committed:
IGNITE-5554 ServiceProcessor may process failed reassignments in timeout thread

glukos committed:
GG-12407: Backport [GG-12364] to 8.1.2 - skipCrc flag added

Jokser committed:
GG-12330 Do not destroy single cache if cache group will be destroyed

Klaster1 committed:
IGNITE-5656 Fixed misaligned checkboxs.

Jokser committed:
IGNITE-5529 Stop exchange worker before cancelling exchange futures.

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

ascherbakoff committed:
Optimize allocations in GridPartitionStateMap

agoncharuk committed:
Merge branch master into ignite-2.1.2

sboikov committed:
Pass io policy in GridMessageListener.

sboikov committed:
Remove some unused parameters.

agoncharuk committed:
IGNITE-5613 - Fixed race on local sequence increment and distributed update

agoncharuk committed:
IGNITE-5613 - Fixed race on local sequence increment and distributed update

ptupitsyn committed:
IGNITE-5628 .NET: Fix jvm.dll lookup paths for JRE

glukos committed:
GG-12408: Backport [GG-12385] and reproducing tests to 8.1.2

dgovorukhin committed:
ignite-2.1.2 more assert info in wal manager

dgovorukhin committed:
Merge remote-tracking branch 'professional/ignite-2.1.2' into ignite-2.1.2

sboikov committed:
Reduced amount of debug logging.

akuznetsov-os committed:
IGNITE-5683 Fixed missing fully qualified class names for generated indexed types on Models screen.

agoncharuk committed:
Merge master into ignite-2.1.2

vsisko committed:
ignite-2.1 Fixed minor issues and typos.

gurustron committed:
IGNITE-5532 .NET: Split CacheLinqTest into partial classes

This closes #2226

glukos committed:
GG-12416: Backport [GG-12402] to 8.1.2

akuznetsov-os committed:
ignite-2.1 Added persistent store metrics.

Commit:
ignite-5685 JDBC prepared statement shouldn't clear parameters after execution

ptupitsyn committed:
IGNITE-5629 .NET: CacheConfiguration copy constructor

glukos committed:
Fixed updateCounter when node is removed (backport of 4fedd0fe)

glukos committed:
Added page type check when saving cache metadata (backport of d38fd8d6)

ilantukh committed:
Muted hanging tests.

ilantukh committed:
Muted hanging tests.

agoncharuk committed:
Cosmetic changes

dgovorukhin committed:
Added onCacheGroupStopped callback

isapego committed:
IGNITE-5663: ODBC: Closing cursor do not reset prepared statement anymore

(cherry picked from commit 64c156e)

isapego committed:
IGNITE-5663: ODBC: Closing cursor do not reset prepared statement anymore

dgovorukhin committed:
Minor code cleanup

glukos committed:
Merge branch 'ignite-2.1' into ignite-2.1.2-merge-ea11

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

Commit:
IGNITE-5558 - Added ability to read WAL in standalone mode - Fixes #2174.

Signed-off-by: Alexey Goncharuk <alexey.goncharuk@gmail.com>

sergey-chugunov-1985 committed:
IGNITE-4536 two tests unmuted, one test removed as obsolete (IGNITE-5592) - Fixes #2235.

Signed-off-by: Alexey Goncharuk <alexey.goncharuk@gmail.com>

ilantukh committed:
IGNITE-5693 - Fixed semaphore flaky test - Fixes #2239.

Signed-off-by: Alexey Goncharuk <alexey.goncharuk@gmail.com>

Commit:
IGNITE-5340: Improved IgniteCacheAbstractQuerySelfTest to better test non-standard index names. This closes #2206.

Commit:
IGNITE-5426: JdbcThinConnection.readOnly state propagation. This closes #2200.

sboikov committed:
Reworked cluster activation/deactivation.

sboikov committed:
Merge remote-tracking branch 'origin/ignite-2.1' into ignite-2.1

glukos committed:
Merge branch 'ignite-2.1' into ignite-2.1.2-merge-ea11

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

ptupitsyn committed:
IGNITE-5605 .NET: Inject resources into remote event filters

Commit:
Merge branch 'ignite-2.1' into ignite-2.1.2

sboikov committed:
Diagnostic info for GridDhtTxFinishFuture.

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

alexpaschenko committed:
IGNITE-5187: Improved DynamicIndexAbstractConcurrentSelfTest reliability. This closes #2219.

sboikov committed:
ignite-2.1 Do not use 'compress' flag for GridDhtPartitionsFullMessage send in discovery message.

sboikov committed:
ignite-2.1 Do not use 'compress' flag for GridDhtPartitionsFullMessage send in discovery message.

sboikov committed:
ignite-2.1

isapego committed:
IGNITE-5576: Added Compute::Run() for C++

isapego committed:
IGNITE-5576: Added Compute::Run() for C++

(cherry picked from commit 80c95ff79f344daf1fca3f094733a24bac2a218d)

dgovorukhin committed:
Added onActivate/onDeactivate callbacks for plugins

dgovorukhin committed:
IGNITE-5604 - Expand WAL iterator buffer if record size is greater than current buffer size - Fixes #2244.

Signed-off-by: Alexey Goncharuk <alexey.goncharuk@gmail.com>

glukos committed:
Code style

agoncharuk committed:
Merge remote-tracking branch 'community/ignite-2.1.2-merge-ea11' into ignite-2.1.2

Commit:
Fixed missing SUID.

samaitra committed:
IGNITE-5695 FlinkIgniteSinkSelfTest is failing due to conflicting default test timeout and default flush frequency - Fixes #2241.

Signed-off-by: samaitra <saikat.maitra@gmail.com>

shroman committed:
Removed unnecessary line from the comments.

sboikov committed:
2.1 Removed extra add in pendingMsgs

sergey-chugunov-1985 committed:
IGNITE-5401 Fix in MarshallerConextImpl. This closes #2205.

sboikov committed:
2.1 Unused imports

alexpaschenko committed:
IGNITE-5702: Set CacheWriteSynchronizationMode.FULL_SYNC for predefined CREATE TABLE templates. This closes #2246.

Commit:
IGNITE-4901 Decrease logging level for DataStremer retry

Commit:
Merge branch 'ignite-2.1.2' of https://github.com/gridgain/apache-ignite into ignite-2.1.2

alexpaschenko committed:
IGNITE-5533: Proper index cleanup in H2 database during DROP TALBE. This closes #2167.

ptupitsyn committed:
IGNITE-5441 .NET: Propagate CacheStore exception stack traces to the caller node

This closes #2247

SBBerkov committed:
Fixed "IGNITE-5390 But in IgniteCacheTxStoreSessionWriteBehindCoalescingTest"

Signed-off-by: nikolay_tikhonov <ntikhonov@gridgain.com>

ptupitsyn committed:
IGNITE-5502 .NET: Persistent Store test

sboikov committed:
2.1 Do print stack traces if tx is cancelled on node stop

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

sboikov committed:
2.1 Added remote node pending messages in diagnostic info

sboikov committed:
2.1 Added test.

Commit:
Added missing header to ByteBufferExpander.

vsisko committed:
IGNITE-5710 Change H2 version in depend to Ignite version.

sboikov committed:
ignite-5075 Fixed row size calculation in canUpdateOldRow

vsisko committed:
Merge branch 'ignite-5710' into ignite-2.1.2

Commit:
Merge branch 'ignite-2.1'

dgovorukhin committed:
IGNITE-5520 - Fixed IgniteChangeGlobalStateFailoverTest

sboikov committed:
2.1 More simple tx cancel on node stop

sboikov committed:
Rename ActionData -> CacheActionData.

akuznetsov-os committed:
ignite-2.1 Added map with previous snapshots.

skalashnikov committed:
IGNITE-5483: Added limited support for Java8 LocalDateTime. This closes #2248.

sboikov committed:
Fixed formatting.

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

sboikov committed:
Fixed formatting.

alexpaschenko committed:
IGNITE-5159: DDL example. This closes #2227.

Commit:
Merge remote-tracking branch 'upstream/ignite-2.1.2' into ignite-2.1.2

ptupitsyn committed:
IGNITE-5717 .NET: Reduce MemoryPolicyConfiguration.MaxSize for persistence tests as a workaround for OOM on default settings

sboikov committed:
2.1 Fixed assert (cacheName can be null here).

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

sk0x50 committed:
ignite-5562: assert statements were changed to the 'if' blocks

skalashnikov committed:
IGNITE-5204: Fixed NPE on certain data with index inlining. This closes #2115.

sboikov committed:
Removed unused method GridDhtPartitionTopology.checkEvictions().

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

skalashnikov committed:
IGNITE-5204: SQL: fixed incorrect partition calculation in case of unicast optimization, when WHERE clause argument type was different from key or affinity key type. This close #2107.

isapego committed:
IGNITE-5582: Implemented Compute::Broadcast for C++

isapego committed:
IGNITE-5582: Implemented Compute::Broadcast for C++

(cherry picked from commit fa974286e8f066a8d6aa57519edf5ec7761be095)

sboikov committed:
Fixed cache plugin validation.

ilantukh committed:
IGNITE-5701 - Some nodes have partitionUpdateCounter equal to 0 after rebalancing

akuznetsov-gridgain committed:
ignite-2.1 Minor.

vsisko committed:
IGNITE-5369 Added support for _key and _val names on QueryEntity XML and code generation.

ptupitsyn committed:
IGNITE-5716 .NET: Fix 2-byte field offset handling

ptupitsyn committed:
.NET: Remove unused import and redundant cast

ptupitsyn committed:
.NET: Fix IgniteConfigurationTest

Jokser committed:
GG-12466 - Clean up public API for snapshots

sboikov committed:
ignite-5446 Alway use late affinity assignment mode

ilantukh committed:
IGNITE-5694 - Added retries for getCollection in DataStructuresProcessor - Fixes #2240.

Signed-off-by: Alexey Goncharuk <alexey.goncharuk@gmail.com>

sboikov committed:
Fixed NPE in setOwners.

sboikov committed:
Merge remote-tracking branch 'community/ignite-2.1.2' into ignite-2.1.2

# Conflicts:
#	modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtPartitionTopologyImpl.java

ptupitsyn committed:
IGNITE-5491 .NET: PersistentStoreMetrics

sboikov committed:
GridDhtPartitionsExchangeFuture minor: removed duplicated updateAlives call.

ptupitsyn committed:
IGNITE-5716 .NET: Fix 2-byte field offset handling - improve tests

Jokser committed:
GG-12418 - WAL hangs on any error during segment rollover

agoncharuk committed:
Fixed NPE when test kernal context is used

sboikov committed:
ignite-5075 Rename cacheId -> groupId

NSAmelchev committed:
IGNITE-5087 Enum comparison fails after marshal-unmarshal with BinaryMarshaller.

Commit:
IGNITE-5679: Example for thin JDBC driver. This closes #2232.

sboikov committed:
Removed unused method.

ptupitsyn committed:
IGNITE-2492 .NET: Peer assembly loading - add TestCacheGetOnRemoteNode

agoncharuk committed:
Added partition release future timing

artemmalykh committed:
IGNITE-5688: Fix SparseDistributedMatrixTest::testCacheBehaviour crash. This closes #2272

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+    private static final long TEST_TIMEOUT = 60_000;
+
+    /** */
+    private static final int NODES_CNT = 3;
```
NODES

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+    private static final int PUT_CNT = 10_000;
+
+    /** {@inheritDoc} */
+    @Override protected int gridCount() {
```
remove method

dream-x commented on modules/core/src/main/java/org/apache/ignite/internal/processors/timeout/GridTimeoutProcessor.java:
```diff
@@ -131,6 +130,24 @@ public void removeTimeoutObject(GridTimeoutObject timeoutObj) {
+     * Wraps {@link TimeoutWorker} in a {@link IgniteThread}
+     * with access to {@link TimeoutWorker#cancel()} method.
+     */
+    private static class GridWorkerWrapper extends IgniteThread{
```
private static class GridWorkerWrapper extends IgniteThread {

dream-x commented on modules/core/src/main/java/org/apache/ignite/internal/processors/timeout/GridTimeoutProcessor.java:
```diff
@@ -273,4 +290,4 @@ public void removeTimeoutObject(GridTimeoutObject timeoutObj) {
     }
-}
\ No newline at end of file
+}
```
new line after `}`

dream-x replied:
https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-A.ConfigureIntelliJIDEAcodestyle

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+    public void testTxRollbackOnStop() throws Exception {
+        IgniteEx ignite1 = grid(0);
+        IgniteEx ignite2 = grid(1);
+        IgniteEx ignite3 = startGrid(gridCount()); //to be closed.
```
```
for (int i = 0; i < NODES; i++)
     startGrid(i);
```
or
```
startGridsMultiThreaded(NODES);
```

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+                throw e;
+        }
+
+        ignite3.close();
```
```
stopGrid(2);

awaitPartitionMapExchange();

info("Ignite3 closed.");
```

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+            info("Tx2 started");
+
+            for (int i = 0; i < PUT_CNT; i++) {
+                assertEquals(cache2.get(i).intValue(), Integer.MAX_VALUE);
```
assertEquals(cache2.get(i), Integer.MAX_VALUE);

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+
+        logTimeoutObjectsFrequency();
+
+        //Check that the values have not changed and lock can be acquired.
```
`// Check that the values have not changed and lock can be acquired.`

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+
+        info("Tx2 stopped");
+
+        //Check that that changes committed.
```
// Check that that changes committed.

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+
+        //Check that that changes committed.
+        for (int i = 0; i < PUT_CNT; i++)
+            assertEquals(cache1.get(i).intValue(), i);
```
assertEquals(cache1.get(i), i);

BiryukovVA replied:
This does not work.

dream-x replied:
sorry :)

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+        IgniteEx ignite2 = grid(1);
+        IgniteEx ignite3 = startGrid(gridCount()); //to be closed.
+
+        IgniteCache<Integer, Integer> cache1 = ignite1.cache(DEFAULT_CACHE_NAME);
```
```
IgniteCache<Integer, Integer> cache1 = grid(0).cache(DEFAULT_CACHE_NAME);
IgniteCache<Integer, Integer> cache2 = grid(1).cache(DEFAULT_CACHE_NAME);
IgniteCache<Integer, Integer> cache3 = grid(2).cache(DEFAULT_CACHE_NAME);
```

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+        logTimeoutObjectsFrequency();
+
+        //Check that the values have not changed and lock can be acquired.
+        try (Transaction tx2 = ignite2.transactions().txStart(PESSIMISTIC, SERIALIZABLE)) {
```
`try (Transaction tx2 = grid(1).transactions().txStart(PESSIMISTIC, SERIALIZABLE)) {`

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+        logTimeoutObjectsFrequency();
+
+        info("Tx1 started");
+        ignite3.transactions().txStart(PESSIMISTIC, SERIALIZABLE, 1000, PUT_CNT);
```
`grid(2).transactions().txStart(PESSIMISTIC, SERIALIZABLE, 1000, PUT_CNT);`

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/timeout/GridTimeoutProcessorSelfTest.java:
```diff
@@ -608,4 +626,28 @@ public void testTimeoutSameEndTime() throws Exception {
+            }
+        });
+
+        ignite.close();
```
stopGrid(0);

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+ */
+public class IgniteTxRollbackOnStopTest extends GridCacheAbstractSelfTest {
+    /** */
+    private static final long TEST_TIMEOUT = 60_000;
```
remove

dream-x commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+
+    /** {@inheritDoc} */
+    @Override protected long getTestTimeout() {
+        return TEST_TIMEOUT;
```
return 60_000;

BiryukovVA commented on modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxRollbackOnStopTest.java:
```diff
@@ -0,0 +1,194 @@
+            info("Tx2 started");
+
+            for (int i = 0; i < PUT_CNT; i++) {
+                assertEquals(cache2.get(i).intValue(), Integer.MAX_VALUE);
```
This does not work.

dream-x replied:
sorry :)

Commit:
IGNITE-5662 Primary index name should contain type ID or name - Fixes #2275.

Signed-off-by: Alexey Goncharuk <alexey.goncharuk@gmail.com>

nva committed:
Web Console: Fixed logic on cluster lost.

Jokser committed:
IGNITE-5684 - Fixed update sequence handling in GridDhtPartitionFullMap

gvvinblade committed:
IGNITE-5686 Endless partition eviction during node shutdown

dgovorukhin committed:
GG-12449 - Fix eviction and checkpoint write race - Fixes #2277.

Signed-off-by: Alexey Goncharuk <alexey.goncharuk@gmail.com>

Commit:
Merge branch 'ignite-2.1'

# Conflicts:
#	modules/platforms/dotnet/Apache.Ignite.Core.Tests/Apache.Ignite.Core.Tests.csproj

ptupitsyn committed:
IGNITE-5446 Always use late affinity assignment mode - update .NET part

sboikov committed:
Exchange future cleanup, added special tasks for reassign/force rebalance.

ptupitsyn committed:
Merge branch 'ignite-2.1'

sboikov committed:
ignite-5446 Only lateAffinity logic in CacheAffinitySharedManager.

SharplEr commented on modules/core/src/test/java/org/apache/ignite/internal/processors/timeout/GridTimeoutProcessorSelfTest.java:
```diff
@@ -608,4 +626,26 @@ public void testTimeoutSameEndTime() throws Exception {
+                    Thread.sleep(10_000);
+                }
+                catch (InterruptedException ignore) {
+                    // Eat exception.
```
Лучше // No-op.

BiryukovVA committed:
IGNITE-5452:
1) Handled eaten InterruptedException in TimeoutObject.
2) Test correctness of rollback a transaction with timeout during the grid stop.

BiryukovVA committed:
Merge remote-tracking branch 'upstream/master' into IGNITE-5452
